compiler/test/fail_compilation/test22765.d

```diff
@@ -0,0 +1,14 @@
+// https://issues.dlang.org/show_bug.cgi?id=22765
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/test22765.d(14): Error: template instance `test22765.Template!null` internal compiler error: C++ `null` template value parameter is not supported
```

Comment:
Internal compiler errors should never be reachable, so this PR introduced a regression.